### PR TITLE
Deep link the referral claim

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change adds support for the referral claim URL deep link. 

## Testing Instructions
1. Check the Referrals feature flag is on
2. Click on this link on the device https://pocket-casts-main-development.mystagingwebsite.com/redeem-guest-pass/secret-referral-code
3. ✅ Verify the claim screen is shown to the user
4. Click on this link https://pocket-casts-main-development.mystagingwebsite.com
5. ✅ Verify the website is opened and the app isn't
6. Click on this link https://pocket-casts-main-development.mystagingwebsite.com/get
7. ✅ Verify the app is opened
8. Turn off the Referrals feature flag
9. Click on this link on the device https://pocket-casts-main-development.mystagingwebsite.com/redeem-guest-pass/secret-referral-code
10. ✅ Verify the claim screen isn't shown to the user

## Screenshots or Screencast 

https://github.com/user-attachments/assets/99a6b097-35e0-4f54-92d7-6430572b146f

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 1]
```diff
@@ -1340,6 +1345,13 @@ class MainActivity :
         }
     }
 
+    private fun openReferralClaim(code: String) {
+        // TODO decide where to store the referral code
+        Timber.i("Referral code: $code")
```
Comment (geekygecko) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:1350]: @ashiagr I wasn't sure where to save the referral code. Maybe we can do this in another PR.

File: modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactory.kt [hunk 2]
```diff
@@ -41,15 +41,16 @@ class DeepLinkFactory(
...
+        PocketCastsWebsiteReferralAdapter(webBaseHost),
         PodloveAdapter(),
         SonosAdapter(),
         ShareListAdapter(listHost),
         ShareListNativeAdapter(),
         SubscribeOnAndroidAdapter(),
         AppleAdapter(),
         CloudFilesAdapter(),
-        UpdageAccountAdapter(),
+        UpgradeAccountAdapter(),
```
Comment (geekygecko) [inline modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactory.kt:53]: Minor typo fix

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          3 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>PocketCastsWebsiteGetAdapter</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>PocketCastsWebsiteReferralAdapter</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>UpgradeAccountAdapter</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactoryTest.kt [hunk 3]
```diff
@@ -245,14 +245,25 @@ class DeepLinkFactoryTest {
...
+            .setData(Uri.parse("https://pocketcasts.com/get"))
 
         val deepLink = factory.create(intent)
 
-        assertEquals(PocketCastsWebsiteDeepLink, deepLink)
+        assertEquals(PocketCastsWebsiteGetDeepLink, deepLink)
+    }
+
+    @Test
+    fun pocketCastsWebsiteGetSlashDeepLink() {
```
Comment (MiSikora) [inline app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactoryTest.kt:259]: Nice! Could you also add a test for a link with query parameters? Context: https://github.com/Automattic/pocket-casts-android/pull/2559

File: modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt [hunk 4]
```diff
@@ -244,6 +244,10 @@ data class SignInDeepLink(
     }
 }
 
+data class PocketCastsWebsiteReferralDeepLink(
+    val code: String,
+) : DeepLink
```
Comment (MiSikora) [inline modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt:249]: Do you think this would should be `UriDeepLink`? This way it could be used for sharing. If we add there are tests like `SignInDeepLinkTest` to be added.

File: modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt [hunk 5]
```diff
@@ -244,6 +244,10 @@ data class SignInDeepLink(
     }
 }
 
+data class PocketCastsWebsiteReferralDeepLink(
```
Comment (MiSikora) [inline modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt:247]: Names seems a bit wordy. Maybe just `ReferralsDeepLink`?

**Review by MiSikora — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -117,6 +117,24 @@
...
+                    android:pathPrefix="/redeem-guest-pass" />
+            </intent-filter>
+            <intent-filter android:autoVerify="true">
+                <action android:name="android.intent.action.VIEW"/>
+                <category android:name="android.intent.category.DEFAULT"/>
+                <category android:name="android.intent.category.BROWSABLE"/>
+                <data
+                    android:host="pocketcasts.com"
+                    android:scheme="https"
+                    android:pathPrefix="/redeem-guest-pass" />
```
Comment (MiSikora) [inline app/src/main/AndroidManifest.xml:136]: Should we make this future-proof and allow query parameters? See: https://github.com/Automattic/pocket-casts-android/pull/2559

File: app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactoryTest.kt (hunk 3 above)
Comment (MiSikora) [reply app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/deeplink/DeepLinkFactoryTest.kt:259]: And maybe another one for a referral link without the code?

File: modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt (hunk 5 above)
Comment (geekygecko) [reply modules/services/deeplink/src/main/kotlin/au/com/shiftyjelly/pocketcasts/deeplink/DeepLink.kt]: Good point, I have renamed this.
